schema/commonmeta/commonmeta.schema.json

```diff
@@ -0,0 +1,541 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://commonmeta.org/commonmeta_v0.9.1.json",
+    "title": "Commonmeta v0.9.1",
+    "description": "JSON representation of the Commonmeta schema.",
+    "additionalProperties": false,
+    "definitions": {
+        "affiliations": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "id": {
+                        "type": "string",
+                        "format": "uri"
+                    },
+                    "name": {
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "name"
+                ]
+            },
+            "uniqueItems": true
+        },
+        "latitude": {
+            "type": "number",
+            "minimum": -90,
+            "maximum": 90
+        },
+        "longitude": {
+            "type": "number",
+            "minimum": -180,
+            "maximum": 180
+        },
+        "geoLocationPoint": {
+            "type": "object",
+            "properties": {
+                "pointLongitude": {
+                    "$ref": "#/definitions/longitude"
+                },
+                "pointLatitude": {
+                    "$ref": "#/definitions/latitude"
+                }
+            },
+            "required": [
+                "pointLongitude",
+                "pointLatitude"
+            ]
+        }
+    },
+    "type": "object",
+    "properties": {
+        "id": {
+            "description": "The unique identifier for the resource.",
+            "type": "string",
+            "format": "uri",
+            "pattern": "^https?://"
+        },
+        "type": {
+            "description": "The type of the resource.",
+            "enum": [
+                "Article",
+                "Audiovisual",
+                "BookChapter",
+                "BookSeries",
+                "Book",
+                "Component",
+                "Dataset",
+                "Dissertation",
+                "Document",
+                "Grant",
+                "Instrument",
+                "JournalArticle",
+                "JournalIssue",
+                "JournalVolume",
+                "Journal",
+                "PeerReview",
+                "PhysicalObject",
+                "ProceedingsArticle",
+                "ProceedingsSeries",
+                "Proceedings",
+                "ReportComponent",
+                "ReportSeries",
+                "Report",
+                "Software",
+                "Other"
+            ]
+        },
+        "additional_type": {
+            "description": "The additional type of the resource.",
+            "type": "string"
+        },
+        "url": {
+            "description": "The URL of the resource.",
+            "type": "string",
+            "format": "uri"
```

Comment:
How does this differ from `id` and `content_url`?

Reply:
`id` is the identifier for the resource, ideally using a persistent identifier such as a DOI, but could also be a identifiers.org URL. The URL can be different, and is for example used internally (e.g. for DOI registrations or to link services that don't use PIDs such as social media).

Reply:
Ah right, got it. Presumably for those cases where the identifier is a CURIE (and can be remapped using identifiers.org), you are using the full `http://identifiers.org/<id>` form, rather than the CURIE form, since the schema has `"format": "uri"` for the `id` field?

We have one DB where there are unique identifiers for data files, but they don't map to a URI -- there is path information but it refers to an internal tape-based storage system. For that reason, and because within (some of?) the bioinformatics community CURIEs are in common usage, KBCMS allows URIs or CURIEs in the `id` field.

Reply:
That makes sense. Will look into adding CURIE support to commonmeta.

Reply:
There are a number of python libs that support CURIEs using existing mapping services - e.g. [`curies`](https://curies.readthedocs.io/en/latest/)

Reply:
Thanks. I am most familiar with identifiers.org as a registry, but will think about how to approach this. Also, need to think about commonmeta-py being based on JSON schema and not a linked data application. So conceptually closer to e.g. frictionless data.